feat: support `output.generatedCode.symbols`

IWANABETHATGUY opened the pull request:
implemented https://rollupjs.org/configuration-options/#output-generatedcode-symbols
### TL;DR

Added support for the `generatedCode.symbols` option to control Symbol.toStringTag usage in namespace objects.

### What changed?

- Added a new `GeneratedCodeOptions` type with `symbols` and `preset` properties
- Updated the `render_namespace_markers` function to use the `symbols` option when determining whether to include Symbol.toStringTag
- Modified CJS, IIFE, and UMD format renderers to pass the generated code options
- Added tests for the new functionality
- Updated CLI help text to include the new options

### How to test?

You can test this feature by setting the `generatedCode.symbols` option in your Rolldown config:

```
export default {
  output: {
    generatedCode: {
      symbols: true
    }
  }
}
```

When enabled, namespace objects will include `Symbol.toStringTag` with a value of 'Module'.

### Why make this change?

This change adds support for controlling whether Symbol.toStringTag is used in namespace objects, which is a feature available in other bundlers. Using Symbol.toStringTag can help with debugging by providing better object representation when inspecting modules, but it might not be desired in all environments, especially those targeting older browsers.lghttps://rollupjs.org/configuration-options/#output-generatedcode-symbolslg

IWANABETHATGUY commented:
* **#6335** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6335?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6335?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6335).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

shulaoda reviewed (approved):
Perhaps we should add some related descriptions? This isn’t a blocker.

IWANABETHATGUY commented:
> Perhaps we should add some related descriptions? This isn’t a blocker.

Done

IWANABETHATGUY committed:
feat: support `output.generatedCode.symbols` (#6335)

implemented https://rollupjs.org/configuration-options/#output-generatedcode-symbols
### TL;DR

Added support for the `generatedCode.symbols` option to control Symbol.toStringTag usage in namespace objects.

### What changed?

- Added a new `GeneratedCodeOptions` type with `symbols` and `preset` properties
- Updated the `render_namespace_markers` function to use the `symbols` option when determining whether to include Symbol.toStringTag
- Modified CJS, IIFE, and UMD format renderers to pass the generated code options
- Added tests for the new functionality
- Updated CLI help text to include the new options

### How to test?

You can test this feature by setting the `generatedCode.symbols` option in your Rolldown config:

```
export default {
  output: {
    generatedCode: {
      symbols: true
    }
  }
}
```

When enabled, namespace objects will include `Symbol.toStringTag` with a value of 'Module'.

### Why make this change?

This change adds support for controlling whether Symbol.toStringTag is used in namespace objects, which is a feature available in other bundlers. Using Symbol.toStringTag can help with debugging by providing better object representation when inspecting modules, but it might not be desired in all environments, especially those targeting older browsers.lghttps://rollupjs.org/configuration-options/#output-generatedcode-symbolslg
